### ISSUE TITLE
AlMgSiMC

### ISSUE THREAD

**Issue by qingyang0315** [davidkleiven/GPAWTutorial#1]
It is really an awesome project. It is really helpful for us to understand if there is a help document. The needed files like ".h5" and ".json“ files are obtained by which python file?

**Comment by davidkleiven** [comment]
Hi, this isn't really a normal project. This is merely a collection of scripts I have written during my PhD, and is primarily intended for personal use (e.g. I just used GitHub for "syncing" files between my desktop at home and the one on work, as well as other computing facilities).

Most of these scripts are connected to the [CLEASE package](https://gitlab.com/computationalmaterials/clease), so if there are particular things you are interested in, I am happy to guide you in the right direction, but I don't recommend using scripts in this repository directly, albeit some of them may be a good starting point for your own scripts.

**Comment by qingyang0315** [comment]
Thanks for your reply. I am very interested in the Phasefield modelling precipitate process. I have learned phase field modelling solidification of multicompnent alloys (liquid to solid transformation). Now I focus on the precipitation process. I have read your paper "Precipitate formation in aluminium alloys: Multi-scale modelling approach". It is very helpful for me. I have learn clease software by myself. If it is convenient for you, can you give an instruction for phase field model？

**Comment by davidkleiven** [comment]
Ok, the phase field model we used in that article is [GOPF](https://github.com/davidkleiven/gopf). GOPF is one possibility, however, later I have realised that a lot of these things are already available in mainstream PF packages. The free and open-source package with the most users, and thus most documented/tested is probably [MOOSE](https://mooseframework.inl.gov/?_ga=2.54594539.116788108.1621691172-905924558.1621691172). MOOSE is a fairly general package, but there is a separate [phase field module](https://mooseframework.inl.gov/modules/phase_field/index.html) and also dedicated modules for nucleation. The most used multiphase models are also available in MOOSE. Have you considered that? I mention another package because it has better support for parallelization (which is at least useful for 3D simulations) and there also exists more examples for MOOSE than GOPF. I also consider using MOOSE more myself.

**Comment by qingyang0315** [comment]
Thanks for your help. I have downloaded  GOPF and learned the examples. I have noticed this code is written with Go language. I have not learned Go. Maybe I should written python code with your framework. In many example files, the output file is binary file? how to convert these binary files to csv files? And in the example "heterogeneousnucleation", I found the result of phi and c are NaN. By the way, I have downloaded Moose, I will try to learn it. Thanks for your help. If it is convenient, can you follow my skype "live:.cid.f84e2b5c89e03253".

**Comment by davidkleiven** [comment]
Alright, I will check the GOPF issue. However, I'm about to finish my degree, so I am only going to maintain GOPF project on a hobby basis.

**Comment by qingyang0315** [comment]
Thanks. Have you successfully compiled the Moose？When I try to compile Moose on Ubuntu, It always shows that "PETSc has not been detected". However, I have successfully installed PETSc  version=3.13. Is there any detail I should mention?

**Comment by davidkleiven** [comment]
Yes, I did compile it, but I recall that there was some hazzle with petsc. I think I had to specify the path where petsc is installed.

**Comment by qingyang0315** [comment]
I have specified the path where petsc was installed. But there is still the error "/home/yq/projects/moose/framework/build/header_symlinks/Moose.h:236:2: error: #error PETSc has not been detected, please ensure your environment is set up properly then rerun the libmesh build script and try to compile MOOSE again.
"
